lib/ui/common/RoundedBackgroundText.dart

```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+
+class RoundedBackgroundTextViewModel {
+  String title;
+  IconData icon;
+
+  RoundedBackgroundTextViewModel({this.title, this.icon});
+}
+
+RoundedBackgroundTextViewModel buildBackgroundTextViewModel({String title, IconData icon}) {
+  return RoundedBackgroundTextViewModel(title: title, icon: icon);
+}
+
+abstract class RoundedBackgroundTextStyle {
+  final Color backgroundColor;
+
+  final BorderRadius borderRadius;
+
+  final EdgeInsets edgePadding;
+  final TextStyle titleTextStyle;
+
+  final int maxLines;
+
+  RoundedBackgroundTextStyle(this.backgroundColor, this.borderRadius,
+      this.edgePadding, this.titleTextStyle, this.maxLines); //X
+
+}
+class _DefaultStyle implements RoundedBackgroundTextStyle {
+
+  final BorderRadius borderRadius = const BorderRadius.all(Radius.circular(20.0));
+  final Color backgroundColor = const Color(0x1425df0c);
+  final EdgeInsets edgePadding = const EdgeInsets.only(top: 8.5, right: 12, left: 12, bottom: 8);
+  final TextStyle titleTextStyle = const TextStyle(fontSize: 15, fontWeight: FontWeight.normal, color: Color(0xff25df0c));
+
+  final int maxLines = 1;
+  const _DefaultStyle();
+}
+
+
+class RoundedBackgroundText {
+  static build(
+  {RoundedBackgroundTextStyle style = const _DefaultStyle(),
+  String title, IconData icon}) {
+    return _buildRoundedBackgroundText(style, title: title, icon: icon);
+  }
+
+  static Widget _buildRoundedBackgroundText(RoundedBackgroundTextStyle style, {String title, icon}) {
+    String defaultValue = "0";
```

Comment:
what's default value? this function should pass in the viewModel, mock the viewModel outside the widget for testing. Davids demo widget will help with this.